api/src/services/users/create_api_key.py

```diff
@@ -59,6 +84,45 @@ def create_api_key(db_session: db.Session, user_id: UUID, json_data: dict) -> Us
     return api_key
 
 
+def _import_api_key_to_aws_gateway(api_key: UserApiKey) -> None:
+    """Import an API key to AWS API Gateway and associate it with a usage plan"""
+    try:
+        config = ApiGatewayConfig()
+
+        gateway_response = import_api_key(
+            api_key=api_key.key_id,
+            name=api_key.key_name,
+            description=f"API key for user {api_key.user_id}",
+            enabled=api_key.is_active,
+            usage_plan_id=config.default_usage_plan_id,
+        )
+
+        if config.default_usage_plan_id:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway and associated with usage plan",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                    "usage_plan_id": config.default_usage_plan_id,
+                },
+            )
+        else:
+            logger.info(
+                "Successfully imported API key to AWS API Gateway (no usage plan configured)",
+                extra={
+                    "api_key_id": api_key.api_key_id,
+                    "gateway_key_id": gateway_response.id,
+                },
+            )
+
+    except Exception as e:
+        # Re-raise as a domain-specific exception without additional logging
+        # since the AWS adapter already logs the underlying error
+        raise ApiGatewayIntegrationError(
+            f"Failed to import API key to AWS API Gateway: {str(e)}"
+        ) from e
```

Comment:
The `from e` bit should attach the stacktrace/error stuff to the error and not require adding it to the message of the error.

I appreciate you adding through error handling and logging, but we can really keep things simple, we don't need to parse exceptions in our try-excepts in most cases. The API's error logic should just handle adding all the contents of an error to our logs.